File: FreshVienna/FTOPMLReader.m
//
//  FTOPMLReader.m
//  FreshVienna
//
//  Created by Thibaut on 5/13/14.
//  Copyright (c) 2014 Thibaut. All rights reserved.
//

#import "FTOPMLReader.h"
#import "FTOPMLItem.h"
@interface FTOPMLReader()
{
    NSMutableArray *loadedUrls;
}
@end

@implementation FTOPMLReader

-(NSArray*)loadUrl:(NSString*)url
{
    loadedUrls = [NSMutableArray array];
    NSXMLParser *xmlparser = [[NSXMLParser alloc] initWithContentsOfURL:[NSURL fileURLWithPath:url]];
    [xmlparser setDelegate:self];
    [xmlparser setShouldResolveExternalEntities:NO];
    // now parse the document
    [xmlparser parse];
    xmlparser = nil;
    return loadedUrls;
}


#pragma mark XML Parser

-(void)parser:(NSXMLParser *)parser didStartElement:(NSString *)elementName namespaceURI:(NSString *)namespaceURI qualifiedName:(NSString *)qName attributes:(NSDictionary *)attributeDict {
    // add code here to load any data members
    // that your custom class might have
    
    if (attributeDict != nil)
    {
        if ([attributeDict objectForKey:@"xmlUrl"])
        {
            FTOPMLItem *item = [[FTOPMLItem alloc] init];
            item.xmlUrl = [attributeDict objectForKey:@"xmlUrl"];
            item.title = [attributeDict objectForKey:@"text"] ? [attributeDict objectForKey:@"text"] : item.xmlUrl;
            [loadedUrls addObject:item];
        }
    }
    
}



// error handling
-(void)parser:(NSXMLParser *)parser parseErrorOccurred:(NSError *)parseError {
    NSLog(@"XMLParser error: %@", [parseError localizedDescription]);
}

-(void)parser:(NSXMLParser *)parser validationErrorOccurred:(NSError *)validationError {
    NSLog(@"XMLParser error: %@", [validationError localizedDescription]);
}


@end

File: FreshSwiftVienna/FTOPMLReader.swift
//
//  FTOPMLReader.swift
//  FreshVienna
//
//  Created by Thibaut on 6/3/14.
//  Copyright (c) 2014 Thibaut. All rights reserved.
//

import Foundation

class FTOPMLReader: NSObject, NSXMLParserDelegate
{
    var loadedUrls:FTOPMLItem[] = [];

    func loadUrl(url:String) -> FTOPMLItem[]
    {
        loadedUrls = [];
        var parser:NSXMLParser? = NSXMLParser(contentsOfURL:NSURL.fileURLWithPath(url));
        parser!.delegate = self;
        parser!.shouldResolveExternalEntities = false;
        
        // now parse the document
        parser!.parse();
        parser = nil;
        return loadedUrls;
    }
    
    
    func parser(parser: NSXMLParser!, didStartElement elementName: String!, namespaceURI: String!, qualifiedName qName: String!, attributes attributeDict: NSDictionary!)
    {
        
        if (attributeDict != nil)
        {
            if attributeDict.objectForKey("xmlUrl")
            {
                var item = FTOPMLItem(xmlUrl: attributeDict.objectForKey("xmlUrl") as String, title:attributeDict.objectForKey("text") as? String);
                loadedUrls.append(item);
            }
        }
    }
}